pkg/stub/handler.go

```diff
@@ -862,13 +891,67 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 		if srcfg != nil {
 			switch srcfg.Spec.ManagementState {
 			case operatorsv1api.Removed:
-				logrus.Debugln("Ignoring secret event because samples resource is in removed state")
-				return nil
+				// so our current recipe to switch to rhel is to
+				// - mark mgmt state removed
+				// - after that complete, edit again, mark install type to rhel and mgmt state to managed
+				// but what about the secret needed for rhel ... do we force the user to create the secret
+				// while still in managed/centos state?  Even with that, the "removed" action removes the
+				// secret the operator creates in the openshift namespace since it was owned/created by
+				// the operator
+				// So we allow the processing of the secret event while in removed state to
+				// facilitate the switch from centos to rhel, as necessitating use of removed as the means for
+				// changing from centos to rhel since  we allow changing the distribution once the samples have initially been created
+				logrus.Printf("processing secret watch event while in Removed state; deletion event: %v", event.Deleted)
```

Comment:
should probably have a `case Managed:` and `default` for completeness (just logging what flow you went through)

Reply:
ok